dcurve/src/internal/clr.ts

```diff
@@ -124,26 +124,28 @@ export class CLR {
     const currentGrantMatch = getGrantMatch(grantId, distribution);
 
     // calculate predicted distribution for each predictionPoint
-    predictionPoints.forEach(async (predictionPoint) => {
-      // calculate distribution with anon contribution
-      const newDistribution: GrantsDistribution = await calcAlgo({
-        // add anon contribution of value predictionPoint
-        contributions: addAnonymousContribution(grantId, { ...grantRoundContributions }, predictionPoint),
-        trustBonusScores: trustBonusScores,
-        // allow for overrides to set calc algo
-        ...options,
-      } as CLRArgs);
-
-      // retrieve only this grantId from the predicted distribution
-      const predictedGrantMatch = getGrantMatch(grantId, newDistribution);
-
-      // push the prediction for the predictionPoint
-      predictions.push({
-        predictionPoint: predictionPoint,
-        predictedGrantMatch: predictedGrantMatch,
-        predictionDiff: predictedGrantMatch - currentGrantMatch,
-      } as GrantPrediction);
-    });
+    void (await Promise.all(
```

Comment:
I think you could do this instead, not that it really matters though

```suggestion
    void Promise.all(
```

Reply:
I think we need to await here because the map has side-effects (collating the predictions)

Reply:
Ah got it, sounds good!